### PR TITLE
feat(typescript): add TypeScript examples

### PR DESCRIPTION
**Pull request by janvennemann**
Full example of TypeScript usage in both Classic and Alloy apps! 🚀

### CODE REVIEW

**Comment by janvennemann** [comment]
@hansemannn I'll add tslint tomorrow and then this should be ready ;)

File: templates/typescript-alloy-app/.gitignore [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/i18n
+build.log
+build
+npm-debug.log
+tmp
+.map
+.project
+.settings
+Thumbs.db
+plugins/ti.alloy
```
Comment (hansemannn) [inline templates/typescript-alloy-app/.gitignore]: We should not put that on .gitignore as it's causing a build failure when missing this file. It should reaaaally not love in the every project anyway, but until thats fixed, let's restore the dir and remove it from here (ryming like boss).

**Review by hansemannn — CHANGES_REQUESTED** [review]
Only one quick note. Works reaally nice and fast!

**Comment by hansemannn** [comment]
Oh, also, you should let them know to run `npm i` before using the setup. I just ran into that whe n testing classic (which also works fine afterwards)! **EDIT**: Added via https://github.com/appcelerator/titanium-boilerplates/pull/2

**Review by hansemannn — APPROVED** [review]
Looks **awesome**!

**Pull request merged**